common/src/model_deployment/deploy.py

```diff
@@ -234,31 +221,85 @@ def main(args):
         ),
         egress_public_network_access=args.egress_public_network_access,
     )
+
     try:
+        print(f"Creating endpoint {endpoint_name}")
+        ml_client.begin_create_or_update(endpoint).wait()
+    except Exception as e:
+        error_msg = f"Error occured while creating endpoint - {e}"
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    try:
+        print(f"Creating deployment {deployment}")
         ml_client.online_deployments.begin_create_or_update(deployment).wait()
     except Exception as e:
         error_msg = f"Error occured while creating deployment - {e}"
         logging.error(error_msg)
         raise Exception(error_msg)
 
-    print("Deployment done with name ", deployment.name)
+    print(f"Deployment successful. Updating endpoint to take 100% traffic for deployment {deployment_name}")
 
     # deployment to take 100% traffic
     endpoint.traffic = {deployment.name: 100}
     try:
         ml_client.begin_create_or_update(endpoint).wait()
+        endpoint = ml_client.online_endpoints.get(endpoint.name)
     except Exception as e:
         error_msg = f"Error occured while updating endpoint traffic - {e}"
         logging.error(error_msg)
         raise Exception(error_msg)
 
+    return endpoint, deployment
+
+
+def main(args):
+    """Run main function."""
+    ml_client = get_ml_client()
+    # get registered model id
+    model_info = {}
+    with open(args.registration_details) as f:
+        model_info = json.load(f)
+    model_id = model_info["id"]
+    model_name = model_info["name"]
+
+    # make sure underscores and slashes are replaced by hyphens and convert them to lower case
+    endpoint_name = re.sub('[/_ ]', '-', model_name)
+    endpoint_name = f"{endpoint_name.lower()}-{int(time.time())}"
+    endpoint_name = endpoint_name[:32].lower()
```

Comment:
A long model name could cause this to be non-unique. Consider adding the time suffix after truncating the endpoint name. Also, no need to lowercase the endpoint name twice.